packages/utils/src/metrics.js

```diff
@@ -0,0 +1,129 @@
+// @ts-check
+import get from 'lodash/get';
+import Fingerprint from 'fingerprintjs2';
+
+const RTC_CHANNEL_NAME = 'endpass-metrics';
+const CANDIDATE_ATTR_REG = /^a=candidate/;
+const CANDIDATE_ATTR_SHORT_REG = /^c=/;
+const ADDRESS_REG = /([0-9]{1,3}(\.[0-9]{1,3}){3}|[a-f0-9]{1,4}(:[a-f0-9]{1,4}){7})/g;
+const GOOGLE_STUN_SERVER = 'stun:stun.l.google.com:19302?transport=udp';
+
+/**
+ * Extracts network address from SDP parameters
+ * @param {string} rawSdp Raw SDP parameters
+ * @returns {string|null}
+ */
+const getAddressFromSDP = rawSdp => {
+  let address = null;
+  const sdpParts = rawSdp.split('\r\n');
+
+  sdpParts.forEach(line => {
+    if (CANDIDATE_ATTR_REG.test(line)) {
```

Comment:
`.forEach` to `.reduce`

Reply:
There are problems with typescript. `forEach` is fitting ok